src/database.c

```diff
@@ -203,9 +231,17 @@ static int write_db(const char *index, const char *data)
 		return 2;
 	}
 
-	key.mv_data = (void *)index;
-	key.mv_size = strlen(index);
-	value.mv_data = (void *)data;
+    len = strlen(index);
+    if (len > MAX_PATH_LEN) { /* key size is greater than LMDB key limit */
+        hash = path_to_hash(index);
+        key.mv_data = (void *)hash;
+        key.mv_size = gcry_md_get_algo_dlen(GCRY_MD_SHA256) * 2 + 1;
+    }
+    else {
```

Comment:
Can we please fix this? i.e. `} else {`. I'm not sure what style this conforms to, but it isn't fair on the eyes...

Reply:
I will fix that.